cpp/include/cudf/column/column_device_view.cuh

```diff
@@ -312,6 +314,49 @@ class alignas(16) column_device_view : public detail::column_device_view_base {
     return const_iterator<T>{count_it{size()}, detail::value_accessor<T>{*this}};
   }
 
+  /**---------------------------------------------------------------------------*
+   * @brief Pair iterator for navigating this column
+   *---------------------------------------------------------------------------**/
+  template <typename T, bool has_nulls>
+  using const_pair_iterator =
+      thrust::transform_iterator<detail::pair_accessor<T, has_nulls>, count_it>;
+
+  /**---------------------------------------------------------------------------*
+   * @brief Return a pair iterator to the first element of the column.
+   *
+   * Dereferencing the returned iterator returns a `thrust::pair<T, bool>`.
+   *
+   * If an element at position `i` is valid (or `has_nulls == false`), then
+   * for `p = *(iter + i)`, `p.first` contains the value of the element at `i`
+   * and `p.second == true`.
+   *
+   * Else, if the element at `i` is null, then the value of `p.first` is
+   * undefined and `p.second == false`.
+   *
+   * @throws `cudf::logic_error` if tparam `has_nulls == true` and 
+   * `has_nulls() == false`
+   * @throws `cudf::logic_error` if column datatype and Element type mismatch.
+   *---------------------------------------------------------------------------**/
+  template <typename T, bool has_nulls>
+  const_pair_iterator<T,has_nulls> pair_begin() const {
+    return const_pair_iterator<T, has_nulls>{
```

Comment:
Can't you just call `make_pair_iterator`?

Reply:
I followed the same way `begin` and `end` is coded. besides, `pair_end` will involve an addition `make_pair_iterator(*this)+size()`.
which is better way and why?

Reply:
Generally DRY (don't repeat yourself) is a good guideline. If there's a function with an implementation exactly the same as this, call the function rather than duplicating its implementation, right?

Reply:
Or would it be cleaner to have make_pair_iterator call this?  Or does the combination of pair_begin() and pair_end() make make_pair_iterator obsolete?

Reply:
Yes. I made make_pair_iterator call pair_begin. It makes make_pair_iterator obsolete.

Reply:
removed obsolete make_pair_iterator.